### ISSUE TITLE
(proper title & reproduction pending)

### ISSUE THREAD

**Issue by mks11** [mobxjs/mobx-react-lite#295]
Sorry, but if you have unrelated questions, these are better asked in separate issue with appropriate details.

My question wasn't unrelated. The title of this issue was exactly what I googled for. While I was trying to resolve the exact same issue, I asked the very related question.

_Originally posted by @FredyC in https://github.com/mobxjs/mobx-react-lite/issues/261#issuecomment-657669855_

**Comment by FredyC** [comment]
Please elaborate more on your question in detail.

**Comment by mweststrate** [comment]
@mks please use `observer` as recommended in the docs instead of `useObserver`. 

For any further help, please provide a runable reproduction in code sandbox so that we have something to talk about. If your issue would be the same, there was no reason to ask your question in the first place as the same solutions would apply.

**Comment by mks11** [comment]
Thanks! I was able to bring back the reactivity inside `useObserver` by creating a new object from the observables from my store, ie. something like 
```
useObserver(() => {
    return {
          items : [...someObservableItemsInMyStore]
     }
})
```

**Comment by mweststrate** [comment]
As said I'd recommend against doing that, since

1) you are now cloning data structures unnecessarily
2) this still wont detect changes inside objects inside these data structures. 

I recommend in general to use `observer` instead of `useObserver` to avoid any confusion and unnecessary penalties.

**Comment by mks11** [comment]
Thank you!

**Comment by mks11** [comment]
I think you are referring to this part of the doc,  "In this case, you can omit the wrapping observer, but you need to be sure there is no other observable present in the component." I for some reason thought "the component" here was where it was being finally used and not the custom hook itself.

**Comment by FreeClimb** [comment]
I support the idea of observables since the knockout and mvvm template.
Аnd now in a large project I tried to use mobx-react-lite with Ant Design as widgets. I have the same problem, the Table component is not updated after any changes in the dataSource. I tried changing useObservable to observable, but the problem persists. Here is my example on boxing code

[https://codesandbox.io/s/frosty-sid-jvg3w](Ant Design Table with CRUD in Modal form)

I would appreciate any help that I am doing wrong.
thank
